pkg/3scale/amp/component/system.go

```diff
@@ -499,9 +490,7 @@ func (system *System) MasterApicastSecret() *v1.Secret {
 
 func (system *System) appPodVolumes() []v1.Volume {
 	res := []v1.Volume{}
-	if system.Options.pvcFileStorageOptions != nil {
-		res = append(res, system.FileStorageVolume())
-	}
+	res = append(res, system.fileStorageVolume())
```

Comment:
We now always add a volume, being a PVC-based volume or a Secret-based volume so I tried to generalize the code at this point to just get the appropriate volume based on the provided options